packages/client-python/src/rocketride/core/runtime/process.py

```
@@ -0,0 +1,178 @@
+
+    while asyncio.get_event_loop().time() < deadline:
+        _flush_log()
+
+        # Check if the process has crashed
+        if pid and not _is_pid_alive(pid):
+            _flush_log()
+            raise EngineError(f'Runtime process (PID {pid}) exited before becoming healthy')
+
+        try:
+            async with aiohttp.ClientSession() as session:
+                async with session.get(url, timeout=aiohttp.ClientTimeout(total=2)) as resp:
```

Ruff (0.15.7)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)